src/Haskell/Ide/Engine/Transport/LspStdio.hs

```diff
@@ -148,9 +155,9 @@ run scheduler _origDir plugins captureFp = flip E.catches handlers $ do
         -- haskell lsp sets the current directory to the project root in the InitializeRequest
         -- We launch the dispatcher after that so that the default cradle is
         -- recognized properly by ghc-mod
-        _ <- forkIO $ Scheduler.runScheduler scheduler errorHandler callbackHandler (Just lf)
-        _ <- forkIO   reactorFunc
-        _ <- forkIO $ diagnosticsQueue tr
+        flip labelThread "scheduler" =<< (forkIO $ Scheduler.runScheduler scheduler errorHandler callbackHandler (Just lf))
+        flip labelThread "reactor" =<< (forkIO reactorFunc)
+        flip labelThread "diagnostics" =<< (forkIO $ diagnosticsQueue tr)
```

Comment:
I have a feeling someone (@bubba , @wz1000 ?) used the `race` primitive to make sure that if one died they all did, and we would know for sure there was a problem, rather than some dangling partly performing thing.

Reply:
That happens with in the Scheduler I think, that spawns the ghc and ide threads.
Will add it to the post-merge list to discuss and document the actualy threading model again.